writing-style-guide.md

```diff
@@ -0,0 +1,267 @@
+
+## Contractions
+
+It's generally fine to use common contractions, as long as the intent is clear. Where there may be the possibility for confusion, be explicit, for example, use "it is" or "it has" if using "it's" could lead to confusion. One common error is to use "it's" in the belief that the apostophe is required if used in a possessive manner. This is never the case. For example:
+
+* Incorrect: "Ably persists it's data in memory for two minutes."
+* Correct: "Ably persists its data in memory for two minutes."
+
+## Number words
+
+For small numbers generally you write the number as a word:
+
+* Avoid: "Ably persists messages in memory for 2 minutes."
+* Better: "Ably persists messages in memory for two minutes."
+
+Larger numbers can stay as numerals, or where precision is required.
+
+## Use active voice
+
+In line with directing the developer succinctly, it is recommended to mostly use the active voice. In active voice the *Subject* *Verbs* the *Object*. Remember SVO.
+
+Examples:
+
+* *Active voice:* The man ate the apple.
+* *Passive voice:* The apple was eaten by the man.
+* *Active voice:* The client creates the connection.
+* *Passive voice:* The connection is created by the client.
+* *Active voice:* Ably provides an Asset Tracking SDK.
+* *Passive voice:* An Asset Tracking SDK is provided by Ably.
+
+Active voice is simpler, more direct, and easier to translate. 
+
+The principle here is you want to guide the developer succinctly and accurately, in the most direct way possible. If you drop into passive voice because it feels correct for the situation that's fine, but generally you want to keep things simple and use active voice.
+
+## Paragraph breaks
+
+In technical writing, and especially blog posts, you can be more generous with your use of paragraph breaks, compared to a novel. Paragraph breaks make the text less overwhelming and easier to read. Generally, you want to avoid dense and cramped text because it's harder to read and digest. From the technical writer's perspective dense text is also harder to maintain.
+
+## Parentheses
+
+Generally avoid long sentences that have embedded parentheses as this makes the text harder to read. You can break out the bracketed text into a new sentence or rewrite the text to not require brackets. Sometimes the bracketed material might be better off as a note or tip admonition.
+
+## Avoid vague and cautious language
+
+Avoid words such as would, should, might, and maybe, as it does not inspire confidence to use such language.
+
+Example:
+
+* Avoid: When an inbound message arrives, you might receive it, or you might get an error, or something!
+* Better: When an inbound message arrives, the message callback handler is invoked.
+* Avoid: If you click the button, the alert box will probably be displayed.
+* Better: When you click the `Submit` button, the alert box is displayed.
+
```

Comment:
Again, I don't know if it belongs here, or an a section of its own, but using technical terminology correctly is paramount. The sentiment needs to be that for technical terms, every word has one meaning, and every meaning has one word.

Reply:
I've added a section on this. The idea is the Ably Glossary will help with this. Unfortunately I am yet to write it, but it is planned. Common example: client library SDK, SDK, client library, library, client. By defining our terms, and compiling the comprehensive glossary, we can get some consistency here.